### ISSUE TITLE
Add support for boostrap v3

### ISSUE THREAD

**Issue by pankajparkar** [ng-bootstrap/ng-bootstrap#998]
As bootstrap 4 is in alpha state, I don't want to start development with bootstrap v4. I tried latest alpha.1.0.0-alpha.10 version. But it doesn't seems to be working.

[Plunkr Here](http://plnkr.co/edit/bziHv92jnVcWdMlfcz78?p=preview)

In above plunkr I changed CSS reference to bootstrap version 3.3.7

Angular: 2.1.0
ng-bootstrap: alpha.1.0.0-alpha.10
Bootstrap: 3.3.7

**Expected**

There should be a configurable setting option by which I can set which bootstrap version I want to target. Also please suggest how can I do the same thing in current version.

**Comment by pkozlowski-opensource** [comment]
Sorry, we've made deliberate choice of supporting only Bootstrap 4, even if it is only in alpha. Here is the reasoning:
* Bootstrap 3 development is effectively frozen and only real development is happening on Bootstrap 4: 
* We think that people will move to Bootstrap 4 very soon as development of it accelerated recently and we can see a stable version on the horizon. I believe that promoting Bootstrap 3 now would mean that people need to face big migration in the future (in other words - we assume that effort of migrating from Bootstrap 3->4 will be much bigger as compared to following releases of Bootstrap 4)
* Supporting 2 versions of Bootstrap would mean that we utilize our resources to support CSS framework that is effectively at its EOL - we believe that we can better use this time to work on World class animations, keyboard support and accessibility. Not to mention the fact that static nature of Angular 2 makes is harder to support several template variants (especially if you want to have things like AoT working properly).

In short: we believe that Bootstrap 4 is the way forward and supporting Bootstrap 3 at this point would be wasteful, both for our users and this project.

**Comment by pankajparkar** [comment]
FYI. I'm currently migrating my A1 application to A2, where I was using `bootstrap-theme.css`(which isn't available here in bootstrap v4). That's why I decided to go with bootstrap v3 until they are done with themes. What should I do in this case?

Removing full backward compatibility doesn't make sense to me, as newer version is there in alpha stage.

**Comment by sumpton** [comment]
@pkozlowski-opensource I would likely adopt your team's decisions as well, were I in that position. And, I am anxious to switch to v4.

I just thought I would share our experience. We have been favouring https://github.com/valor-software/ng2-bootstrap for this reason. We have an existing theme, which we purchased from the Bootstrap team -- mostly to support them. It replaced our Metronic theme. So, we are faced with the theme migration anyway. We see switching from an ng2-bootstrap library to another as an additional, difficult refactoring.

We do think about changing to v4 now, but it would be a lot smoother -- for us, were you also supporting v3. I don't expect you to, but it would have been perfect for us.

It is always amazing how long it takes for projects to release, take Angular 2 for example. I laugh at myself for writing the Bootstrap theme guys last March that they should release a version based on beta 4, thinking a release must be "just around the corner" :)

**Comment by pkozlowski-opensource** [comment]
@sumpton I hear you and believe me that we've spent debatting this issue internally a _lot_. Ultimately it boils down to this: shell we spend available resources (time) to invest in the World-class Bootstrap 4 support or move slower and have BS3 support as well. We made a call to go with BS 4 as based on the argument's I've shared as it seems to make more sense in the longer run. We are painfully aware that this might not be the call others would do....

Not sure what to say more. If you _really_ care about BS 3 support and have available time to dedicate please step in - we could see how to accommodate 2 versions (a separate branch etc.). But keep in mind that this is a non-negligible effort as we are talking about code, documentation, supporting users, doing releases etc.

**Comment by pankajparkar** [comment]
@sumpton Correct, you got my point. Today I spent my whole day on how to create `bootstrap-theme` with BS4. After logging [this issue](https://github.com/twbs/bootstrap/issues/21078) to twitter bootstrap page I came to know that it can be configurable via `scss` compilation. As you said it is going to take several days to create `bootstrap-theme` was there in `BS3` version. That's was I was thinking that It would be better to have BS3 support by `ng-bootstrap`. Otherwise I need to take a one step back and may be I'd go for `ng2-boostrap` by `Valor`.

@pkozlowski-opensource Instead of creating to different repo, can't we make `ng-boostrap` configurable to both BS versions. In real scenario for older Enterprise application who wanted to migrate there application from A1 to A2. They can't really take step to upgrade BS3 to BS4. So there it comes to restriction that they can't use `ng-bootstrap`. Since then they will think to move for `ng2-bootstrap` as they have BS3 & BS4 support.

**Comment by sumpton** [comment]
@pkozlowski-opensource you are great! People like me deeply appreciate what you and your contributors do. You tools are amazing, and I respect your decisions. We will all shift to BS4 when we can. I can hardly wait. I want Cards! -- for example.

I did not realize that supporting BS3 would be a lot of work. Thank you for educating me.

It makes more sense for us to invest in changing our theme to BS4 and working with their beta, than making your project support BS3. I get that. I only made my comment because we are so crazy busy on our core project that we greedily grab whatever works now, and likely others do too. I would like to see your project gain as much traction as possible.

Again, thanks!
